cli/src/hooks/init/compatibility.ts

```diff
@@ -0,0 +1,109 @@
+import { Hook } from '@oclif/core';
+import { readFile, stat, writeFile, mkdir } from 'fs/promises';
+import semver from 'semver';
+import path from 'path';
+import fetch from 'node-fetch';
+
+export const ONE_DAY = 1000 * 60 * 60 * 24 * 1;
+
+export type Packages = 'cli' | 'sdk';
+export type Compatibility = {
+  [key in Packages]: {
+    latest: string;
+    compatibility: { range: string }[];
+  };
+};
+export type PackageJson = { dependencies: Record<string, string> };
+
+export const check = async (params: {
+  pkg: 'cli' | 'sdk';
+  currentVersion: string;
+  compatibilityObj: Compatibility;
+}) => {
+  const { pkg, compatibilityObj } = params;
+  const currentVersion = semver.coerce(params.currentVersion)?.version as string;
+  const updateAvailable = semver.lt(currentVersion, compatibilityObj[pkg].latest);
```

Comment:
If the user is using version ^1.1.1 and the newest version is 1.1.2 they will still see an update available log.